src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,312 @@
+                        Close();
+                    }
+                };
+            }
+        }
+
+        public async void Open(uint length = 300)
+        {
+            var toMove = _outterBox.Geometry;
+            toMove.Y = 0;
+
+            await RunMoveAnimation(_outterBox, toMove, length);
+
+            if (!_isOpen)
+            {
+                _isOpen = true;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async void Close(uint length = 300)
+        {
+            var toMove = _outterBox.Geometry;
+            toMove.Y = Geometry.Height - MinimumDrawerHeight;
+
+            await RunMoveAnimation(_outterBox, toMove, length);
+
+            if (_isOpen)
+            {
+                _isOpen = false;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async Task<bool> ShowAsync(EWidget target, Easing easing = null, uint length = 300, CancellationToken cancelltaionToken = default(CancellationToken))
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            await Task.Delay(1000);
+
+            if (cancelltaionToken.IsCancellationRequested)
+            {
+                cancelltaionToken.ThrowIfCancellationRequested();
+            }
+
+            _outterBox.Show();
+            var opacity = _outterBox.Opacity;
+
+            new Animation((progress) =>
+            {
+                _outterBox.Opacity = opacity + (int)((255 - opacity) * progress);
+
+            }).Commit(this, "FadeIn", length: length, finished: (p, e) =>
+            {
+                _outterBox.Opacity = 255;
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        public async Task<bool> HideAsync(EvasObject target, Easing easing = null, uint length = 300)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var opacity = _outterBox.Opacity;
+            new Animation((progress) =>
+            {
+                _outterBox.Opacity = opacity - (int)(progress * opacity);
+
+            }).Commit(this, "FadeOut", length: length, finished: (p, e) =>
+            {
+                _outterBox.Opacity = 0;
+                _outterBox.Hide();
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            _gestureBox = new Box(parent);
+            return _gestureBox.Handle;
+        }
+
+        void Initialize()
+        {
+            AlignmentX = -1;
+            AlignmentY = -1;
+            WeightX = 1;
+            WeightY = 1;
+
+            _gestureBox.RepeatEvents = true;
+
+            _outterBox = new Box(this);
+            _outterBox.Show();
+
+            _contentBox = new Box(this);
+            _contentBox.PropagateEvents = false;
+
+            _label = new ElmSharp.Label(_outterBox)
```

Comment:
It is better to use `Box` instead of `Label`, next time, we will attach image on this

Reply:
Fixed.